packages/cli-hooks/src/check-update.js

```diff
@@ -0,0 +1,314 @@
+  try {
+    currentVersion = await getProjectPackageVersion(packageName);
+    latestVersion = await fetchLatestPackageVersion(packageName);
+    if (hasAvailableUpdates(currentVersion, latestVersion)) {
+      releaseNotesUrl = getReleaseNotesUrl(packageName, latestVersion);
+    }
+  } catch (err) {
+    if (typeof err === 'string') {
+      dependencyError = { message: err };
+    } else if (err instanceof Error) {
+      dependencyError = { message: err.message };
+    }
+  }
+  return {
+    name: packageName,
+    current: currentVersion,
+    latest: latestVersion,
+    update: hasAvailableUpdates(currentVersion, latestVersion),
+    breaking: hasBreakingChange(currentVersion, latestVersion),
+    message: undefined,
+    url: releaseNotesUrl,
+    error: dependencyError,
+  };
+}
+
+/**
+ * Finds the current version of a local project package.
+ * @param {string} packageName - Name of the package to lookup.
+ * @returns {Promise<string>} A stringified semver of the found version.
+ */
+async function getProjectPackageVersion(packageName) {
+  const stdout = await execWrapper(`npm list ${packageName} --depth=0 --json`);
+  const currentVersionOutput = JSON.parse(stdout);
+  if (!currentVersionOutput.dependencies || !currentVersionOutput.dependencies[packageName]) {
+    throw new Error(`Failed to gather project information about ${packageName}`);
+  }
+  return currentVersionOutput.dependencies[packageName].version;
+}
+
+/**
+ * Gets the latest package version.
+ * @param {string} packageName - Package to search for the latest version of.
+ * @returns {Promise<string>} The most recent version of the published package.
+ */
+async function fetchLatestPackageVersion(packageName) {
+  const command = `npm info ${packageName} version --tag latest`;
+  const stdout = await execWrapper(command);
+  return stdout;
+}
+
+/**
+ * Formats the URL with the release notes for a package.
+ * @param {string} packageName - The package with the release.
+ * @param {string} latestVersion - Recent release version.
+ * @returns {string | undefined} A URL with release notes.
+ */
+function getReleaseNotesUrl(packageName, latestVersion) {
+  if (packageName === SLACK_BOLT_SDK) {
+    return `https://github.com/slackapi/bolt-js/releases/tag/@slack/bolt@${latestVersion}`;
+  } if (packageName === SLACK_CLI_HOOKS) {
+    return `https://github.com/slackapi/node-slack-sdk/releases/tag/@slack/cli-hooks@${latestVersion}`;
+  }
+  return undefined;
+}
+
+/**
+ * Checks if the latest version is more recent than the current version.
+ * @param {string | undefined} current - Package version available in the project.
+ * @param {string | undefined} latest - Most up-to-date dependency version available.
+ * @returns {boolean} If the update will result in a breaking change.
+ */
+export function hasAvailableUpdates(current, latest) {
+  if (!current || !latest || !clean(current) || !clean(latest)) {
+    return false;
+  }
+  return gt(latest, current);
```

Comment:
Awesome 💯

Reply:
Thanks! It's so much cleaner. Appreciate the suggestion 🙌